test/extensions/filters/listener/original_dst/original_dst_fuzz_test.cc

```diff
@@ -0,0 +1,72 @@
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.h"
+
+#include "common/network/utility.h"
+
+#include "extensions/filters/listener/original_dst/original_dst.h"
+
+#include "test/extensions/filters/listener/original_dst/original_dst_fuzz_test.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/network/mocks.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace OriginalDst {
+
+DEFINE_PROTO_FUZZER(
+    const envoy::extensions::filters::listener::original_dst::v3::OriginalDstTestCase& input) {
+
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  }
+
+  NiceMock<Network::MockListenerFilterCallbacks> callbacks_;
+  NiceMock<Network::MockConnectionSocket> socket_;
+
+  try {
+    auto address_ = Network::Utility::resolveUrl(input.address());
+    ON_CALL(socket_, localAddress()).WillByDefault(testing::ReturnRef(address_));
+    ON_CALL(callbacks_, socket()).WillByDefault(testing::ReturnRef(socket_));
+
+    if (address_ != nullptr) {
+      ON_CALL(socket_, addressType()).WillByDefault(testing::Return(address_->type()));
+      if (socket_.addressType() == Network::Address::Type::Ip) {
+        ON_CALL(socket_, ipVersion()).WillByDefault(testing::Return(address_->ip()->version()));
+      }
+    }
+
+    auto filter = std::make_unique<OriginalDstFilter>();
```

Comment:
Why not ingest the proto config?

Reply:
Replied to Alex's comment above about config:

> original_dst doesn't actually have any config options that you can set, but I figured that a fuzzer for a generic listener filter (like original_src) would, so I kept the field for consistency. Should I remove it?

The constructor for OriginalDstFilter doesn't accept the proto config, but most other listener filters should (I think).